### PR TITLE
rename the type of the accumulator of fold functions to 'acc

### LINKED ISSUE #11026: Stdlib interface files improvement proposal: the type of an accumulator should always be 'acc
Hello,

I think this would make type signatures in the stdlib way more readable.
Especially since some people have fun at making the accumulator sometimes
as the first parameter, sometimes as the second one, for dubious reasons
(being consistent would be a way better reason for choosing the position of the accumulator
in the parameters list).

Lowering the cognitive burden for users is a laudable thing.

Regards,
F.

### PR DESCRIPTION
**Pull request by v-gb**
Fixes #11026

This is the same sort of changes as #11667, but for the whole stdlib, since @UnixJunkie didn't intend to do more than the List module, and @Octachron wanted everything done before merging.

### CODE REVIEW

**Review by gasche — APPROVED** [review]
Thanks!

Looks okay, but you forgot List.fold_{left,right}2 I believe.

**Comment by v-gb** [comment]
Ah, you're right. I double-checked I didn't miss functions, but still missed this one because the type and function name were on different lines. I'll add them once the tests are done running locally + edit the Changes file.

File: Changes [hunk 1]
```diff
@@ -100,6 +100,10 @@ Working version
   (Greta Yorsh, review by Xavier Leroy, Guillaume Melquiond and
   Gabriel Scherer)
 
+- #11026, #11858: Rename the type of the accumulator of fold functions
+  to 'acc (Valentin Gatien-Baron, report by Francois Berenger, review
+  by Gabriel Scherer)
```
Comment (nojb) [inline Changes:105]: Author + reviewers should go on a separate line.

**Review by nojb — APPROVED** [review]
LGTM

File: Changes (hunk 1 above)
Comment (gasche) [reply Changes:105]: I went ahead and merged, and fixed this in trunk.

**Comment by UnixJunkie** [comment]
@sliquister thanks for this contribution. I add to correct so many times not compiling source code because I did not put the accumulator at the right position...

**Pull request merged**